zipkin-web/src/main/resources/app/js/page/default.js

```diff
@@ -52,12 +52,15 @@ define(
           const endTs = query.endTs || new Date().getTime();
           const serviceName = query.serviceName || '';
           const annotationQuery = query.annotationQuery || '';
+          const queryWasPerformed = serviceName && serviceName.length > 0;
```

Comment:
> What happens if I GET the home page with ?serviceName=x?

The backend performs a query against the storage engine, for traces with serviceName x. So the query is performed.